### ISSUE TITLE
CPU utilization metrics seems to be inconsistent

### ISSUE THREAD

**Issue by kioie** [apache/cloudstack#3471]
##### ISSUE TYPE

 * Bug Report
 
##### COMPONENT NAME

~~~
CPU utilization
~~~

##### CLOUDSTACK VERSION

~~~
4.11+
~~~

##### SUMMARY

CPU utilization metrics seems to be inconsistent with the actual CPU usage in the VM. The error seems to be generated from this `getVMStat` method in the `LibvirtComputingResource` class
```
 /* get cpu utilization */
            VmStats oldStats = null;

            final Calendar now = Calendar.getInstance();

            oldStats = _vmStats.get(vmName);

            long elapsedTime = 0;
            if (oldStats != null) {
                elapsedTime = now.getTimeInMillis() - oldStats._timestamp.getTimeInMillis();
                double utilization = (info.cpuTime - oldStats._usedTime) / ((double)elapsedTime * 1000000);

                final NodeInfo node = conn.nodeInfo();
                utilization = utilization / node.cpus;
                if (utilization > 0) {
                    stats.setCPUUtilization(utilization * 100);
                }
            }

```

##### STEPS TO REPRODUCE

~~~
Run stress test on the VM with 100% cpu utilization then confirm VM Metric on the statistics tab 
~~~
![Screen Shot 2019-07-05 at 10 13 41 AM](https://user-images.githubusercontent.com/20902920/60705523-009b9480-9f10-11e9-9032-5c8e65a193a6.png)

![Screen Shot 2019-07-05 at 10 13 21 AM](https://user-images.githubusercontent.com/20902920/60705622-3e98b880-9f10-11e9-9fae-afdf059c6181.png)

##### EXPECTED RESULTS

~~~
CPU utilization should be equally as high
~~~

##### ACTUAL RESULTS

~~~
CPU utilization is inconsistent i.e 8.4% (as shown)
~~~

**Comment by rhtyd** [comment]
It takes upto the time set by the background thread to kick in. Refer to stats collector related timeout/interval global settings.

**Comment by kioie** [comment]
Okay...so typically, at what point would it be triggered for it to kick in? Been running some stress tests on several VMs for 3.5hrs now, seems to be returning the same metrics

**Comment by rhtyd** [comment]
The vm.stats.interval is typically 60s. As long as your KVM agent is UP and able to process VM stats every 60s, you should see the stats correctly. Did you try against 4.11/latest or master?

I can confirm against master and KVM and get back to you next week. In my local lab I'm unable to see the issue against 4.11.3.0+kvm.

**Comment by kioie** [comment]
To confirm my config for `vm.stats.interval` is also at 60s also on KVM. Incidentally, looking at the code, I think VMWare and Xen are handling this differently, so they may possibly be returning correct results (haven't tested though)

Yes, tested on 4.11.2.0/4.12/master with ubuntu16.04/18.04/Centos7 vms

My stress test ended up running for 5hrs and the metrics never really changed.

**Comment by DaanHoogland** [comment]
@kioie, is this still an issue?

**Comment by DaanHoogland** [comment]
closing this @kioie , feel free to reopen if you encounter the issue again.

**Comment by weizhouapache** [comment]
this is fixed by #4381
